x-pack/legacy/plugins/ml/public/datavisualizer/index_based/components/actions_panel/actions_panel.tsx

```diff
@@ -40,6 +50,23 @@ export const ActionsPanel: FC<Props> = ({ indexPattern }) => {
         </h2>
       </EuiTitle>
       <EuiSpacer size="s" />
+      <div style={recognizerResultsCount === 0 ? { display: 'none' } : {}}>
```

Comment:
Any reason doing this with CSS/display? Usually we mostly do the condition in JSX itself like
```
{recognizerResultsCount !== 0 && <Fragment>...</Fragment}
```

Reply:
it's because the contained `DataRecognizer ` needs to be initialised and have `recognizerResults` passed to it so it can be in control of whether this section is displayed.
It's a bit of a backwards way of working that comes from the fact it was originally an angular component and would trigger an `ng-show`

Reply:
Thanks for clarifying!

Reply:
I added a comment clarifying the usage of `display:none`